src/ckeditor.jsx

```diff
@@ -51,19 +80,29 @@ export default class CKEditor extends React.Component {
 	}
 
 	_initializeEditor() {
-		this.props.editor
-			.create( this.domContainer.current, this._getConfig() )
-			.then( editor => {
-				this.editor = editor;
+		if ( this.context instanceof ContextWatchdog ) {
+			this.watchdog = new ContextWatchdogToEditorWatchdogAdapter( this.context );
```

Comment:
We need to find a better name for that adapter. Couldn't it be `EditorWatchdogAdapter`?

Reply:
Actually it clearly says what it keeps inside and what interface it matches.

Reply:
`AtSomePointWeWillWriteAStoryAsAClassName` because it's _clearly says what it keeps inside and what interface it matches._

I'd suggest to simplify it and add proper docs to the class.